github_helper.py

```diff
@@ -0,0 +1,132 @@
+"""
+github_helper provides several methods to help interface with Github
+"""
+
+import requests
+
+GITHUB_ORG = 'apache'
+SECRET_FILE = '../../github_auth/apache-beam.secret'
+SOURCE_REMOTE = 'github'
+TARGET_BRANCH = 'master'
+TARGET_REMOTE = 'apache'
+
+GITHUB_API_ROOT = 'https://api.github.com'
+GITHUB_REPO_FMT_URL = GITHUB_API_ROOT + '/repos/{0}/{1}'
+
+BOT_NAME = 'merge-bot'
+GITHUB_SECRET = '../../github_auth/mergebot.secret'
+
+
+class GithubHelper:
+  """
+  GithubHelper provides a lightweight authenticated wrapper for the Github API.
+  """
+  def __init__(self, org, repo):
+    self.org = org
+    self.repo = repo
+    with open(GITHUB_SECRET, 'r') as key_file:
+      self.bot_key = key_file.read().strip()
+    print('Loaded key file.')
+
+  def fetch_prs(self):
+    """
+    Gets the PRs for the configured repository
+
+    :return: returns a JSON object containing the pull request if successful,
+             else None.
+    """
+    print('Loading pull requests from Github at {}.'.format(PULLS_URL))
+    r = requests.get(pulls_url(self.org, self.repo),
+                     auth=(BOT_NAME, self.bot_key))
+    if r.status_code != 200:
+      print('Oops, that didn\'t work. Error below,
```

Comment:
I see no wait, nor retry. If we do retry, it should happen within some wrapper around `requests.get` I think.